endgame_postprocessing/post_processing/pipeline.py

```diff
@@ -145,13 +152,30 @@ def pipeline(input_dir, working_directory, pipeline_config: PipelineConfig):
         working_directory, all_iu_data, pipeline_config.disease
     )
 
+    # Selecting the minimum starting year_id that exists for all IUs
+    minimum_year = max(all_iu_data.groupby("iu_name")["year_id"].min().to_list())
+    # Selecting the maximum ending year_id that exists for all IUs
+    maximum_year = min(all_iu_data.groupby("iu_name")["year_id"].max().to_list())
+
     if not pipeline_config.include_country_and_continent_summaries:
         return
 
     country_aggregates = [
         country_aggregate(
             country_composite,
-            all_iu_data[all_iu_data["country_code"] == country_composite["country_code"].values[0]],
+            all_iu_data[
+                (all_iu_data["country_code"] == country_composite["country_code"].values[0]) &
+                (
+                    (
+                        (all_iu_data["year_id"] >= minimum_year) &
+                        (all_iu_data["year_id"] <= maximum_year)
+                    ) |
+                    (
+                        all_iu_data["year_id"].isna()
```

Comment:
❓ How come we need to include na year data? If it is required, then a comment explaining why here please